[DX-1097] Update SessionState protobuf documentation to include BasicAuthData and example hash algorithms (#4324)
TykTechnologies/tyk-docs: tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md

The key is a string that denotes the name of the header, the values are a list of strings that hold the content of the header, this is useful when the header has multiple associated values.
This field is available for Go, Python and Ruby since tyk v5.0.4 and  5.1.1+.

### BasicAuthData (coprocess_session_state.proto)

The `BasicAuthData` contains a hashed password and the name of the hashing algorithm used. This is represented by the `basic_auth_data` attribute in [SessionState](#session-state) message.

> ```yaml

Reviewer: ```suggestion
```json
```

> The `BasicAuthData` contains a hashed password and the name of the hashing algorithm used. This is represented by the `basic_auth_data` attribute in [SessionState](#sessionstate-session_stateproto) message.

Reviewer 2: ```suggestion
The `BasicAuthData` contains a hashed password and the name of the hashing algorithm used. This is represented by the `basic_auth_data` attribute in [SessionState](#session-state) message.
```

"basicAuthData": {

>     "password": "a_hashed_password_presentation",

Reviewer 2: ```suggestion
    "password": <a_hashed_password_presentation>,
```

>     "password": "a_hashed_password_presentation",

Reviewer 2: I think that it's safer to use something that is not valid yaml so users will remember to change it. We use this format throughout the docs.

>     "hash": "the_hashing_algorithm_used_to_hash_the_password"

Reviewer 2: ```suggestion
    "hash": <the_hashing_algorithm_used_to_hash_the_password>
```

}
```

#### Field Descriptions

`password`
A hashed password.

`hash`

> Name of the hashing algorithm used to hash the password, e.g. *bcrypt*, *Argon2*.

Reviewer: ```suggestion
Name of the hashing algorithm used to hash the password.
```

First time ever hearing of argon2, unless we list explicit values, we shouldn't make up examples.

Author: Thanks @titpetric 

I have looked through the source on GHub and found [this](https://github.com/TykTechnologies/tyk/blob/ebe349cbf16e31a96067486291a91893cf5c7a0a/config/config.go#L679)

Is the suggestion below ok?

```suggestion
Name of the hashing algorithm used to hash the password, e.g. *murmur64*, *murmur128*, *sha256* or *bcrypt*
```

Reviewer: I'd prefer you omit the details, and if necessary link https://tyk.io/docs/basic-config-and-security/security/key-hashing/ for more information.

Author: > I'd prefer you omit the details, and if necessary link https://tyk.io/docs/basic-config-and-security/security/key-hashing/ for more information.

Thanks @titpetric I've updated it in the branch as detailed below and will merge and release:

_Name of the [hashing algorithm]({{< ref "basic-config-and-security/security/key-hashing" >}}) used to hash the password._
